docs/hooks/setup/symfony.md

```diff
@@ -0,0 +1,107 @@
+## Overview
+
+The [__Symfony EventDispatcher__](http://symfony.com/components/EventDispatcher) is a
+common event system used by many PHP applications and frameworks.  For
+example, Symfony SE, Drupal 8, Magento, Laravel, CiviCRM, and many others
+support `EventDispatcher`.
+
+In CiviCRM v4.7.19+, you can use Symfony `EventDispatcher` with hooks.
+
+```php
+Civi::dispatcher()->addListener('hook_civicrm_EXAMPLE', $callback, $priority);
+```
+
+Using `EventDispatcher` is useful if you need more advanced features, such as:
+
+ * Setting the priority of an event-listener
+ * Registering an object-oriented event-listener
+ * Registering a dynamic, on-the-fly event-listener
+ * Registering multiple listeners for the same event
+ * Registering for internal/unpublished events
+
+For a general introduction or background on `EventDispatcher`, consult the [Symfony documentation](http://symfony.com/doc/2.7/components/event_dispatcher.html).
+
+## Example: `Civi::dispatcher()`
+
+In this case, we have a CiviCRM extension or Drupal module named `example`.
+During the system initialization, we lookup the `EventDispatcher`, call
+`addListener()`, and listen for `hook_civicrm_alterContent`.
+
+```php
+function example_civicrm_config(&$config) {
+  if (isset(Civi::$statics[__FUNCTION__])) { return; }
+  Civi::$statics[__FUNCTION__] = 1;
+
+  Civi::dispatcher()->addListener('hook_civicrm_alterContent', function($event) {
+    $event->content = 'hello ' . $event->content;
+  });
+}
+```
+
+> __Note__: In some environments, `hook_civicrm_config` runs multiple times. The flag
```

Comment:
Suggestion: use [admonitions](https://docs.civicrm.org/dev/en/latest/markdownrules/#admonitions) here to make it look more consistent with other docs